### PR TITLE
Feat/multiclass segtb

### LINKED ISSUE #549: Refactor igvc_perception NN things
[cnn.py](https://github.com/RoboJackets/igvc-software/blob/master/igvc_perception/src/cnn/cnn.py) currently is a bit of a mess and needs to be cleaned up. Also, anything related to the NN in the `igvc_perception` folder is messy.

AC: Refactored cnn.py and `igvc_perception`

### PR DESCRIPTION
**Pull request by yyajima21**
# Description
This pr refactored and updated a build instruction for the multi-class segmentation package. Additionally, I updated the data preparation pipeline to generate npy files from raw dataset. I added a dataset download path to the readme file.
{{ State the context of this PR }}

This PR does the following:
- Refactors current multi-class segmentation code to remove hard-coded values.
- Adds an updated build instruction with a valid data download path and tensorboard instructions
- Fixes data processing pipeline to generate train and test dataset.

Fixes #549 #718 

# Testing steps (If relevant)
## Test Case 1
1. Install required dependencies using 'pip install -r requirements'
2. Follow the training instruction and run the training script.
3. Follow the tensortboard instruction to visualize the training results.

Expectation: You should see tensorboard application.

# Self Checklist
- [x] I have formatted my code using `make format`
- [x] I have tested that the new behaviour works

### CODE REVIEW

**Comment by yyajima21** [comment]
Regarding the tensorboard implementation pr #718, the catalyst package already contains tensorbaord logger by default. So I don't think we need to implement additional tensorboard to the train.py.

**Comment by yyajima21** [comment]
@matthewhannay567 @andrewyarovoi  Do you have any updates?

File: igvc_perception/src/multiclass_segmentation/README.md [hunk 1]
```diff
@@ -20,9 +46,21 @@ Using transfer learning, we can simultaneously classify lines and barrels in ima
 ## Build Instructions 
 
 1. Download the dependencies in requirements.txt manually or using `pip install -r requirements.txt`. 
-2. Obtain a dataset containing the images (PNGs) and masks (JSONs). Skip steps #2 and #3 if using .npy files. 
+2. Obtain a dataset containing the images (PNGs) and masks (JSONs) [here](https://cloud.robojackets.org/apps/files/?dir=/RoboNav/Software/dataset/robonav_dataset&fileid=351245). Skip steps #2 and #3 if using .npy files.
```
Comment (matthewhannay567) [inline igvc_perception/src/multiclass_segmentation/README.md:49]: I think these should be `#3` and `#4`.

**Review by matthewhannay567 — CHANGES_REQUESTED** [review]
It works, just one small change before approving.

File: igvc_perception/src/multiclass_segmentation/README.md (hunk 1 above)
Comment (yyajima21) [reply igvc_perception/src/multiclass_segmentation/README.md]: Thank you for your review! I fixed the issue now.

**Review by matthewhannay567 — APPROVED** [review]
Looks good!

**Pull request merged**